src/Swift.Runtime/src/Swift/Runtime/ISwiftObject.cs

```diff
@@ -27,10 +27,15 @@ public interface ISwiftObject
     /// <summary>
     /// Gets the protocol conformance descriptor for the given type
     /// </summary>
-    public static abstract ProtocolConformanceDescriptor GetProtocolConformanceDescriptor<TProtocol>() where TProtocol : class;
+    public static abstract ProtocolConformanceDescriptor GetProtocolConformanceDescriptor<TProtocol>() where TProtocol : ISwiftProtocol;
 }
 
-/// <summary>  
+public interface ISwiftProtocol
```

Comment:
I am not sure about this:
```csharp
public interface IProtocol
{
    public static abstract string GetSomething();
}

public interface IPrintable : IProtocol
{
    static string IProtocol.GetSomething() => "IPrintable";
}

public interface IWritable : IProtocol
{
    static string IProtocol.GetSomething() => "IWritable";
}

public class SomeClass : IPrintable, IWritable
{
// I dont think this will compile
}
```

Also this would make any class implementing a protocol be a `ISwiftProtocol` as in `val is ISwiftProtocol` returns true

Reply:
I agree. Reverted from this PR.